ostd/src/mm/heap_allocator/slab_allocator/mod.rs

```diff
@@ -152,38 +152,45 @@ impl Heap {
     }
 
     /// Allocates a chunk of the given size with the given alignment. Returns a pointer to the
-    /// beginning of that chunk if it was successful. Else it returns `Err`.
+    /// beginning of that chunk and remaining bytes in buddy system allocator if it was successful.
+    /// Else it returns `Err`.
+    ///
     /// This function finds the slab of lowest size which can still accommodate the given chunk.
     /// The runtime is in `O(1)` for chunks of size <= 4096, and `O(n)` when chunk size is > 4096,
-    pub fn allocate(&mut self, layout: Layout) -> Result<usize, AllocError> {
-        match Heap::layout_to_allocator(&layout) {
+    pub fn allocate(&mut self, layout: Layout) -> Result<(usize, usize), AllocError> {
```

Comment:
Change the return type to `Result<(*mut u8, usize), AllocError>` to make it clear that the first field inside `Ok(_)` is the pointer. The current caller converts the return value from `usize` to `*mut u8` anyway.

To be consistent with `allocate`, the `deallocate` method should be modified to take `*mut u8` instead of `usize`.